__tests__/postService.test.js

```diff
@@ -33,21 +34,27 @@ beforeAll(() => {
         };
     });
     postDAO.getPost.mockImplementation(async (id) => {
-        for (let i = 0; i < mockDatabase.length; i++){
-            if (mockDatabase[i].itemID == id){
+        for (let i = 0; i < mockDatabase.length; i++) {
+            if (mockDatabase[i].itemID == id) {
                 return {
                     $metadata: {
                         httpStatusCode: 200
                     },
-                    Item: mockPost1
+                    Item: mockDatabase[i]
+                };
+            } else {
+                return {
+                    $metadata: {
+                        httpStatusCode: 400
+                    }
                 };
             }
         }
         return false;
```

Comment:
Instead of false, I would have this return a status code of 200 but no item. The reason being that when I use getPostById in PostService for testing, I get a status code of 500 instead of 400.

Reply:
The 500 is probably because `throwIfError` takes the response from DynamoDB and throws a generic 500 if the response has a status of 400 or higher, or a response is falsy. It should definitely return a 200 and no item, since that's what the actual DAO would do in this case.